### PR TITLE
CF-2381: Input errors adjustments

### PR DESCRIPTION
**Pull request by bkozera**
[CF-2381](https://jira.spirenteng.com/browse/CF-2381)
---
Errors for Input should be shown under Input field (currently above in Winterfell). 
In addition,  if there are errors Input should receive hasError prop so it can by styled accordingly.

This will work nicely with https://github.com/mudynamics/apoc-react/pull/190 and https://github.com/mudynamics/cyberflood_react/pull/210

### CODE REVIEW

File: dist/question.js [hunk 1]
```diff
@@ -118,20 +118,24 @@ var Question = (function (_React$Component) {
...
-          placeholder: this.props.input.placeholder,
-          required: this.props.input.required,
-          classes: this.props.classes,
-          onChange: this.handleInputChange.bind(this, this.props.questionId),
-          onBlur: this.handleInputBlur.bind(this, this.props.questionId),
-          onKeyDown: this.props.onKeyDown
-        }, extraprops)),
+        React.createElement(
+          Input,
+          _extends({ name: this.props.questionId,
```
Comment (ihor-yamshchykov) [inline dist/question.js]: I suggest using Object.assign instead, I would rather remove lodash

Comment (bkozera) [inline dist/question.js]: I think this whole code is not needed - it looks like it's extending default object..doesn't make sense if you use `defaultProps`.
Anyway - this is far from finished, I only pushed what I have done so far (linting question.jsx and button.jsx)

Comment (bkozera) [inline dist/question.js]: sorry the label was wrong :D

Comment (bkozera) [inline dist/question.js]: @ihor-yamshchykov this code in dist is transpiled - we don't have direct control over it. We could probably change some settings to use `Object.assign`...
But in general it all should be refactored, I just wanted to make a small change for now.

Comment (ihor-yamshchykov) [inline dist/question.js]: missed that it's dist sorry

File: src/question.js [hunk 2]
```diff
@@ -137,8 +136,11 @@ class Question extends React.Component {
                  onChange={this.handleInputChange.bind(this, this.props.questionId)}
                  onBlur={this.handleInputBlur.bind(this, this.props.questionId)}
                  onKeyDown={this.props.onKeyDown}
+                 hasError={!!validationErrors.length}
                  {...extraprops}
-          />
+          >
+          {validationErrors}
```
Comment (dmalts) [inline src/question.js]: indent this properly, please <3

Comment (bkozera) [inline src/question.js]: this whole lib needs adjusting...here's the branch (in progress): https://github.com/mudynamics/Winterfell/pull/4

Comment (dmalts) [inline src/question.js]: good

**Comment by dmalts** [comment]
I've linked this feature branch and tested it along with https://github.com/mudynamics/cyberflood_react/pull/210. Everything works as expected, except i have this Error in console:

screen shot 2017-03-27 at 15 06 58

**Comment by bkozera** [comment]
@dmalts This error comes when you use npm link with Winterfell. Looks like there are mulitiple react version imported (one inside the winterfell and the other one we use in CF). This works file when you remove linking. 
We should look into this but for now I would not worry about this. Maybe you can create an issue?

**Comment by dmalts** [comment]
@bkozera sure, here - https://jira.spirenteng.com/browse/CF-2396

**Pull request merged**